src/command-parser/parsers.ts

```diff
@@ -293,15 +297,18 @@ const parseNativeScriptData = (data: any): NativeScript => {
       }
     case NativeScriptType.ALL:
     case NativeScriptType.ANY:
-      if (!data.scripts || !Array.isArray(data.scripts)) {
+      if (!('scripts' in data) || !data.scripts || !Array.isArray(data.scripts)) {
         throw Error(Errors.InvalidNativeScriptFile)
       }
       return {
         type,
         scripts: data.scripts.map(parseNativeScriptData),
       }
     case NativeScriptType.N_OF_K:
-      if (!isCorrectNumber(data.required) || !data.scripts || !Array.isArray(data.scripts)) {
+      if (!('required' in data) || !isCorrectNumber(data.required)) {
+        throw Error(Errors.InvalidNativeScriptFile)
+      }
+      if (!('scripts' in data) || !data.scripts || !Array.isArray(data.scripts)) {
```

Comment:
`!.data.scripts` may evaluate to `true` if scripts is an empty array - is that desired?

Reply:
I'm not sure. It probably does not make much sense to have an empty array, but it seems the Ledger app accepts this, so I won't change it.